todo_app/app.py

```diff
@@ -2,34 +2,46 @@
 from dotenv import load_dotenv, find_dotenv
 from flask import Flask, render_template, request, redirect
 from todo_app.view_models.homepage_view_model import HomepageViewModel
-from todo_app.network.trello_client import TrelloClient
-from todo_app.data.trello_items import ItemProvider
+from todo_app.network.mongodb_client import MongoDBClient
+from todo_app.data.mongodb_items import ItemProvider
 
 from todo_app.flask_config import Config
 
-def create_item_provider():
-    trello_client = TrelloClient(
-        os.getenv('TRELLO_BOARD_ID'),
-        os.getenv('TRELLO_API_BASE_URL'),
-        os.getenv('TRELLO_API_KEY'),
-        os.getenv('TRELLO_API_TOKEN')
+def create_item_provider(environment):
+    mongodb_client = MongoDBClient(
+        os.getenv('DATABASE_CONNECTION_STRING'),
+        environment
     )
 
-    return ItemProvider(trello_client)
+    return ItemProvider(mongodb_client)
 
-def create_app(env_path='.env'):
+def get_env_path(environment):
+    if environment == 'test':
+        return 'env/.env.test'
+    else:
+        # Currently development and production are using the same credentials
+        # Which is obviously bad but we will change this later
+        return '.env'
+
+def load_dotenv_for_environment(environment):
+    env_path = get_env_path(environment)
     load_dotenv(find_dotenv(env_path))
 
+def create_app(environment='development'):
+    print("Running the app...")
+    print(f"Environment: {environment}")
+    load_dotenv_for_environment(environment)
+
     app = Flask(__name__)
     app.config.from_object(Config())
 
-    item_provider = create_item_provider()
+    item_provider = create_item_provider(environment)
 
     @app.route('/')
     def index():
         items = item_provider.get_items()
```

Comment:
Nice - that this is such a simple change in this file is a really good sign that you picked an effective interface originally 😄